brd-ios/breadwallet/src/Redesign/Views/MainSwapView.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  See the LICENSE file at the project root for license information.
+//
+
+import UIKit
+import SnapKit
+
+struct MainSwapConfiguration: Configurable {
+    var shadow: ShadowConfiguration?
+    var background: BackgroundConfiguration?
+}
+
+struct MainSwapViewModel: ViewModel {
+    var fromFiatAmount: String
+    var fromCryptoAmount: String
+    var toFiatAmount: String
+    var toCryptoAmount: String
+}
+
+class MainSwapView: FEView<MainSwapConfiguration, MainSwapViewModel> {
+    private lazy var containerStackView: UIStackView = {
+        let stack = UIStackView()
+        stack.axis = .vertical
+        stack.distribution = .fill
+        return stack
+    }()
+    
+    private var topSwapCurrencyView = SwapCurrencyView()
+    private var bottomSwapCurrencyView = SwapCurrencyView()
+    
+    private lazy var dividerWithButtonView: UIView = {
+        let view = UIView()
+        
+        let lineView = UIView()
+        lineView.backgroundColor = LightColors.Outline.two
+        
+        view.addSubview(lineView)
+        lineView.snp.makeConstraints { make in
+            make.height.width.equalTo(1)
+            make.leading.trailing.equalToSuperview()
+            make.center.equalToSuperview()
+        }
+        
+        let buttonView = FEButton()
+        buttonView.setImage(UIImage(named: "SwitchPlaces"), for: .normal)
+        buttonView.addTarget(self, action: #selector(switchPlacesButtonTapped), for: .touchUpInside)
+        
+        view.addSubview(buttonView)
+        buttonView.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+        }
+        
+        return view
+    }()
+    
+    var didChangeFromFiatAmount: ((SwapMainModels.Amounts.CurrencyData?) -> Void)?
+    var didChangeFromCryptoAmount: ((SwapMainModels.Amounts.CurrencyData?) -> Void)?
+    var didChangeToFiatAmount: ((SwapMainModels.Amounts.CurrencyData?) -> Void)?
+    var didChangeToCryptoAmount: ((SwapMainModels.Amounts.CurrencyData?) -> Void)?
+    
+    private var subviewOrder: Int = 0
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        configure(background: config?.background)
+        configure(shadow: config?.shadow)
+    }
+    
+    override func setupSubviews() {
+        super.setupSubviews()
+        
+        content.addSubview(containerStackView)
+        containerStackView.snp.makeConstraints { make in
+            make.top.bottom.leading.trailing.equalToSuperview()
+        }
+        
+        containerStackView.addArrangedSubview(topSwapCurrencyView)
+        containerStackView.addArrangedSubview(dividerWithButtonView)
+        containerStackView.addArrangedSubview(bottomSwapCurrencyView)
+        
+        dividerWithButtonView.snp.makeConstraints { make in
+            make.height.equalTo(32)
+        }
+        
+        (containerStackView.arrangedSubviews.first(where: { $0 is SwapCurrencyView }) as? SwapCurrencyView)?
```

Comment:
couldnt u just add a tappedCallback on ur SwapCurrencyViews?
or at least not filter through subviews, cause u already have a reference to both CurrencyViews